### DISCUSSION TITLE
Not sure if this an issue with LG IP Control module, or WebOS quirk?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [WesSouza/lgtv-ip-control#120] [Q&A]
Not sure if this an issue with LG IP Control module, or WebOS quirk? The following via lgtv-ip-control commands:

1. Set TV to HDMI1
2. Set screenmuteon
3. Set TV to HDMI2
4. HDMI1 is displayed... this does make sense?

IMHO. expected result is allmuteoff state, and TV displays HDMI2, however, TV displays HDMI1, as set in step 1. So although allmuteoff is the new screen mute state, the HDMI2 set input never seems to happen. Either the WebOS is not honoring HDMI2 or lgtv-ip-control is not?

The issue is masked, if you do the following...

1. Set TV to HDMI1
2. Set screenmuteon
3. Set TV to HDMI1
4. HDMI1 displayed

Never see the changed input being dishonored, because original input and new input is same.

The issue never happens if you do...

1. Set TV to HDMI1
2. Set screenmuteon
3. Set allmuteoff
4. HDMI1 displayed

The issue is also masked depending on the actual TV remote, some models of TV remotes have direct input for HDMI1, HDMI2, etc. In my case, not having direct input buttons, my remote seems to send allmuteoff when I hit the input button. And I have to hit input again to select a different input. So maybe the above behavior is WebOS design?

**Comment by WesSouza** [comment]
That’s hard to say, in theory the library waits the TV to respond an OK after commands, if you await the command.

The screen mute was added by @jadoc, I’m not sure my current TV model supports it (I switched to an LCD model), so it’s hard to test.

Can you add a delay before switching to HDMI2?

Inside an `async` function you can:

```js
await new Promise((resolve) => setTimeout(resolve, 2000));
```

**Comment by Jibun-no-Kage** [comment]
I tried a couple of things. a delay, then sending the input again.  But I am still playing with it.  I suspect this really is a quirk with the WebOS implementation.  Not a big deal really, just seemed odd.

**Comment by Jibun-no-Kage** [comment]
Closing this, since it seems to be an WebOS API design issue or quirk
